docs/setup/docker.asciidoc

```diff
@@ -78,14 +78,16 @@ services:
 [[environment-variable-config]]
 ==== Environment variable configuration
 
-Under Docker, Kibana can be configured via environment variables. When
+Under Docker, {kib} can be configured via environment variables. When
 the container starts, a helper process checks the environment for variables that
 can be mapped to Kibana command-line arguments.
 
 For compatibility with container orchestration systems, these
 environment variables are written in all capitals, with underscores as
 word separators. The helper translates these names to valid
-Kibana setting names.
+{kib} setting names.
+
+WARNING: All information that you include in environment variables is visible through the `ps` command, so only include nonsensitive information.
```

Comment:
> so only include nonsensitive information.

I'm not a docker expert _at all_, but it's my understanding that using environment variables for secrets is tolerable if you understand and are willing to accept the risks.

I think we should also tell our users how to securely provide sensitive configuration options if they don't wish to use environment variables. For this, I expect we'll want to guide them to using a [keystore](https://www.elastic.co/guide/en/kibana/current/secure-settings.html), but I'm not sure of the proper way to get the keystore into the docker container. @elastic/kibana-operations, do you have any suggestions here? It is as simple as copying a pre-configured keystore to the config directory? Is this directory `/usr/share/kibana/config/`?

Reply:
yep you got it, there's two approaches - mount a volume containing the keystore or build an image on top with the configuration changes copied in.

1) `docker --mount source=kibanana-config target=/usr/share/kibana/config/`
1) ```
FROM docker.elastic.co/kibana/kibana:7.7.0
ADD kibana_keystore.yml /usr/share/kibana/config/
```

roughly psuedocode here, don't have the exact syntax offhand.

with that said i'd push for keeping these separate as a best practice.  native docker/vault secrets as a first approach to solving environment variables.  keystore if we need disk obfuscation.

Reply:
@legrego would you be okay with us circling back on the keystore in a separate issue?  this is mostly coming from the readiness of it, afaik there's no password protection yet

Reply:
> @legrego would you be okay with us circling back on the keystore in a separate issue? this is mostly coming from the readiness of it, afaik there's no password protection yet

Yep, I'm ok with that, given the change @KOTungseth made in [https://github.com/elastic/kibana/pull/67623/commits/669fd8d039681b1729b865ac61ee31d8074c1e14 ](https://github.com/elastic/kibana/pull/67623/commits/669fd8d039681b1729b865ac61ee31d8074c1e14)